miscelanea/004-route-traffic/README.md

```diff
@@ -0,0 +1,520 @@
+poc-004-3-services   9/9     Running   0          80s   10.143.0.205   permanent-bdd7p-worker-x7cmm   <none>           <none>
+```
+
+Execute the checking script:
+
+```shell
+./check-scenario-3.sh
+```
+
+And finally we delete the resources allocated for the PoC by:
+
+```shell
+APP=poc-004-3 make app-delete
+```
+
+## Scenario 4 - External access
+
+Different options have been investigated for the external access:
+
+- **NodePort**: This needs some hacks in OpenStack by adding some changes to the
+  security groups.
+- **externalIPs + NodePort**: This investigation is moved to a new ticket
+  to avoid this ticket grow too much.
+- **externalIPs + LoadBalancer**: This investigation is moved to a new ticket
+  to avoid this ticket grow too much.
+
+### nodePort
+
+[poc-004-4c.yaml](poc-004-4c.yaml).
+
+> You need to modify the security group of your OpenStack infrastructure
+> to allow communication to the NodePorts indicated into the manifest files.
+> One problem detected here is that the rules created by default doesn't
+> allow to route the traffic to inside the cluster because the remote prefix
+> should be `0.0.0.0/0` instead of `10.8.0.0/16`.
+
+We can see the rules as the below in yaml format:
+
+```yaml
+# openstack security group show <cluster-infra-id>-worker
+rules: created_at=''2020-11-17T19:45:20Z'', direction=''ingress'', ethertype=''IPv4'',
+  id=''bf4db301-ea8d-4cc9-bb66-69f0b012e201'', port_range_max=''32767'', port_range_min=''30000'',
+  protocol=''udp'', remote_ip_prefix=''0.0.0.0/0'', updated_at=''2020-11-17T19:45:20Z''
+
+  created_at=''2020-11-17T19:44:43Z'', direction=''ingress'', ethertype=''IPv4'',
+  id=''7d561c9b-b80c-4bc2-b32c-d09f99d05012'', port_range_max=''32767'', port_range_min=''30000'',
+  protocol=''tcp'', remote_ip_prefix=''0.0.0.0/0'', updated_at=''2020-11-17T19:44:43Z''
+```
+
+Options to this situation:
+
+- Add the rule by hand in OpenStack after the OpenShift cluster has been
+  deployed.
+
+- idm-ocp-provisioner could create a security group and add it as an
+  additional security group in the `install-config.yaml`:
+
+  ```yaml
+  compute:
+  - name: worker
+    platform:
+      openstack:
+        additionalSecurityGroupIDs:
+        - 7ee219f3-d2e9-48a1-96c2-e7429f1b0da7
+  ```
+
+Now we can see that the ports are opened, after deploying the manifest:
+
+```raw
+# APP=poc-004-4c make app-deploy
+# nmap 10.0.135.189 -p 31080,31443,31389,31636,31088,31464
+Starting Nmap 7.80 ( https://nmap.org ) at 2020-11-17 21:47 CET
+Nmap scan report for poc-05-c.apps.permanent.idmocp.lab.eng.rdu2.redhat.com (10.0.135.189)
+Host is up (0.17s latency).
+
+PORT      STATE SERVICE
+31080/tcp open  unknown
+31088/tcp open  unknown
+31389/tcp open  unknown
+31443/tcp open  unknown
+31464/tcp open  unknown
+31636/tcp open  unknown
+
+Nmap done: 1 IP address (1 host up) scanned in 0.36 seconds
+```
+
+We can play with the proof of concept by:
+
+```shell
+APP=poc-004-4c make app-deploy
+./check-scenario-4c.sh
```

Comment:
Add here the output from the check-scenario-4c.sh script execution.

Reply:
I have updated the output for a failure scenario and for a successful scenario. Also I have added how to create the rule for the security group in openstack.